graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/GraphQLClient.kt

```diff
@@ -40,7 +40,12 @@ fun interface RequestExecutor {
     fun execute(url: String, headers: Map<String, List<String>>, body: String): HttpResponse
 }
 
-data class HttpResponse(val statusCode: Int, val body: String?)
+data class HttpResponse(val statusCode: Int, val body: String?) {
+    var headers = mapOf<String, List<String>>()
+    constructor(statusCode: Int, body: String?, headers: Map<String, List<String>>) : this(statusCode, body) {
```

Comment:
So this constructor is to keep backwards compatibility with existing Java users? 
Would be good to test that explicitly.

Reply:
yes, to avoid breaking existing api.

Reply:
will add a test case.

Reply:
Added a separate test case for the response headers.